pymc_marketing/mmm/media_transformation.py

```diff
@@ -0,0 +1,278 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Module for applying media transformations to media data.
+
+Examples
+--------
+Create a media transformation for online and offline media channels:
+
+.. code-block:: python
+
+    from pymc_marketing.mmm import (
+        GeometricAdstock,
+        HillSaturation,
+        MediaTransformation,
+        MichaelisMentenSaturation,
+    )
+
+    offline_media_transform = MediaTransformation(
+        adstock=GeometricAdstock(l_max=15),
+        saturation=HillSaturation(),
+        adstock_first=True,
+    )
+    online_media_transform = MediaTransformation(
+            adstock=GeometricAdstock(l_max=10),
+            saturation=MichaelisMentenSaturation(),
+            adstock_first=False,
+        ),
+    )
+
+Create a media configurations for offline and online media channels:
+
+.. code-block:: python
+
+    from pymc_marketing.mmm import (
+        MediaConfig,
+        MediaConfigs,
+    )
+
+    media_configs: MediaConfigs = [
+        MediaConfig(
+            name="offline",
+            columns=["TV", "Radio"],
+            media_transformation=offline_media_transform,
+        ),
+        MediaConfig(
+            name="online",
+            columns=["Facebook", "Instagram", "YouTube", "TikTok"],
+            media_transformation=online_media_transform,
+        ),
+    ]
+
+
+Apply the media transformation to media data in PyMC model:
```

Comment:
Can this configuration specification be used through the `media_config` parameter with the `MMM` model class? If so, could you provide an example here as well 🙏 ?

Reply:
There isn't a `media_config` parameter at the moment. But we could make one 👀